include/systems/implicit_system.h

```diff
@@ -303,7 +303,6 @@ class ImplicitSystem : public ExplicitSystem
    */
   inline const SparseMatrix<Number> & get_system_matrix() const
     {
-      libmesh_assert(matrix);
```

Comment:
Why did these need to be removed?

If we don't have a code path that's pulling the system matrix before it's been allocated then this assert is correct.  If we *do* have such a code path then ```return *matrix``` is undefined behavior and this whole function isn't correct.